File: 006/main.go
package main

import (
	"fmt"
	"math"
)


func main(){
	limit := 100
	sum := limit * (limit + 1) /2
	// proof: https://github.com/grantwforsythe/project-euler/tree/main/proofs/006.pdf
	sum_sq := ((2 * limit + 1)*(limit + 1) * limit) / 6
	
	result := math.Pow(float64(sum), 2) - float64(sum_sq)
	fmt.Printf("Answer: %d\n", int(result))
	
}

File: 006/main.py
def main():
    digits = list(range(1,101))
    result = difference(digits) 
    print(f"Answer: {result}")


def difference(digits: list) -> int:
    return square_of_sums(digits) - sum_of_squares(digits)


def square_of_sums(digits: list) -> int:
    return sum(digits)**2


def sum_of_squares(digits: list) -> int:
    return sum(map(lambda x: x**2, digits))


if __name__ == '__main__':
    main()